packages/react/src/components/Facepile/Facepile.base.tsx

```diff
@@ -77,7 +70,7 @@ export class FacepileBase extends React.Component<IFacepileProps, {}> {
               showTooltip,
             )}
           </ul>
-          {overflowButtonProps ? this._getOverflowElement(personasOverflow) : null}
+          {this._getOverflowElement(personasOverflow)}
```

Comment:
I don't think we should make this change as the props are optional in that you don't have to pass them if you don't render an overflow button. However, `overflowButtonProps` are how you attach the `onClick` handler to the overflow button. Without these props the overflow button will be rendered but not actionable.

Reply:
@spmonahan `overflowButtonProps` is passed as a property to `FacepileButton` component in `_getDescriptiveOverflowElement` and `_getIconElement` private methods and removing this check here will not affect in making the overflow button actionable and will only help in making `overflowButtonProps` property as optional.

Let me elaborate a bit.

`_getOverflowElement` method should always be called irrespective of `overflowButtonProps` being passed because, `_getOverflowElement` method takes care of calling `_getDescriptiveOverflowElement` and `_getIconElement` private methods based on the `overflowButtonType` property. If it does not find `overflowButtonType` property, it returns `null`, which is good.

Now, inside these `_getDescriptiveOverflowElement` and `_getIconElement` private methods, `overflowButtonProps` property is passed to `FacepileButton` component and this is where `overflowButtonProps` actually gets applied.

Reply:
> check here will not affect in making the overflow button actionable and will only help in making overflowButtonProps property as optional.

`FacepileButton` is just a [BaseButton](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Facepile/FacepileButton.tsx#L15) with no default `onClick` handler. `overflowButtonProps` is the API for passing this handler in via `Facepile` and without these props I believe it is not possible to attach an `onClick` handler to the overflow button.

With your change if I set `overflowButtonType` to a value other than `none` and do not provide `overflowButtonProps` the overflow button will be rendered and it will do nothing. [While this is also possible to create this situation today](https://codepen.io/seanms/pen/ZEVPvQJ) our examples expressly show setting `onClick`, which is not a perfect solution but it does help put folks on the correct path.

`Facepile` was written before my time on Fluent but my takeaway from all this is that `overflowButtonProps` is required to correctly use the overflow button, but optional in the sense that you do not need to render an overflow button in all `Facepile` use cases.

I can see the motivation for your change (and were it not likely to cause breaks I'd be fine with it) but there is another problem with this change. The line in question was last touched about 5 years ago according to `git blame`:

![facepile_overflowButtonProps_git_blame](https://github.com/microsoft/fluentui/assets/93940821/8b3706a9-86c3-409a-8536-39aa4ffa48d2)

This means that the existing behavior is the expected behavior at this point. Today if I set `overflowButtonType` to something other than "none" but do not pass `overflowButtonProps` an overflow button will not be rendered. After your change a button will be rendered that does nothing, a pretty significant change of behavior 🙂. Other users could reasonably call such a change a bug or regresssion.

My feeling is that this change does make `Facepile`'s API nicer but the existing API is not a serious issue, the behavior is long-standing and a change is likely to break some users. I don't think we should make this change.

Is there a bug this change fixes that cannot be fixed another way?

Reply:
Yeah, updating the docs to describe the actual behavior feels like the right thing. Mind opening a PR for that?

Reply:
Sure, will do.

Would it be okay to assign you as the reviewer for that?

Reply:
Yep!